### DISCUSSION TITLE
Need TT100K 306cls pt/onnx, Chinese traffic sign. Thx!

### DISCUSSION THREAD

**Discussion by jiangxingjx** [ultralytics/ultralytics#22827] [Q&A]
Need TT100K 306cls pt/onnx, Chinese traffic sign. Thx!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jiangxingjx, thank you for your interest in Ultralytics and Chinese traffic sign detection 🚀! This is an automated response to help you get started quickly — an Ultralytics engineer will also review your discussion and assist you soon 😊

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, including how to train, export, and run YOLO models, which may help with TT100K-style datasets and exporting to formats like PyTorch and ONNX.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training or pretrained-model ❓ Question (for example, about TT100K, multi-class traffic sign models, or specific export formats like ONNX), please provide as much information as possible, including:
- Your exact goal (training from scratch, fine-tuning, or just inference)
- Dataset details and examples
- Any logs or export commands you’ve already tried  
and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)
- Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community

## Upgrade

Before proceeding, please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to ensure you have the latest features and fixes:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by PrashantDixit0** [comment]
> Need TT100K 306cls pt/onnx, Chinese traffic sign. Thx!

@glenn-jocher TT-100k dataset looks like a comprehensive dataset for Traffic Sign detection and Classification. It would be useful for traffic-related applications.
https://cg.cs.tsinghua.edu.cn/traffic-sign/

Let me know, If I can take it up and add it to the dataset section. Looking forward to hearing from you

**Reply by glenn-jocher** [reply]
That would be helpful, thanks for offering. If you’d like to add TT100K to the datasets section, please follow the “Contribute New Datasets” steps in the [Datasets overview](https://docs.ultralytics.com/datasets/) and the general guidelines in the [Ultralytics contributing guide](https://docs.ultralytics.com/help/contributing/), and open a small, focused PR (e.g., docs plus example `data.yaml`, no default config changes); we don’t have official TT100K pretrained weights yet, but this would make it much easier for users to train YOLO11 on the dataset.

**Reply by PrashantDixit0** [reply]
Yep, Thanks @glenn-jocher for the heads up. I'll shortly raise a PR for it :+1:

**Reply by glenn-jocher** [reply]
Sounds good, thanks for taking it on—when you open the PR, please include a minimal `data.yaml` and a brief TT100K description (plus any dataset-specific preprocessing notes) so others can easily reproduce training, and we’ll review it once it’s up.

**Reply by PrashantDixit0** [reply]
@jiangxingjx I have added `TT100K.yaml` for direct training
#22871
